docs/book/basics/simple_controller.md

```diff
@@ -1,144 +1,139 @@
-- Watch for events on *Deployment*s and call Reconcile for the key of the Owning ContainerSet
-- Return the `GenericController` from the function
+- To enqueue a Reconcile for the object in the event use a `handler.EnqueueRequestForObject`
+- To enqueue a Reconcile for the object that created the object in the event use a `handler.EnqueueRequestForOwner`
+  with the type of the owner e.g. `&handler.EnqueueRequestForOwner{OwnerType: &appsv1.Deployment{}, IsController: true}`
+- To enqueue Reconcile requests for an arbitrary collection of objects in response to the event, use a
+  `handler.EnqueueRequestsFromMapFunc`.
 
-**Note:** when watching the Deployment, a Predicate is used to filter events where the
-ResourceVersion of the Deployment have not changed.  This is an optimization to filter
-out Deployment events that don't require a reconcile.
+Example:
+
+- Create a new `ContainerSetController` struct that will.
+  - Invoke Reconcile with the Name and Namespace of a *ContainerSet* for *ContainerSet* create / update / delete events
+  - Invoke Reconcile with the Name and Namespace of a *ContainerSet* for *Deployment* create / update / delete events
 
 #### Reference
 
-- See the [controller libraries](https://godoc.org/github.com/kubernetes-sigs/kubebuilder/pkg/controller) godocs
-for reference documentation on watches.
+- See the [controller libraries](https://godoc.org/sigs.k8s.io/controller-runtime/pkg) godocs for reference
+documentation on the controller libraries.
 - See the [controller code generation tags](https://godoc.org/github.com/kubernetes-sigs/kubebuilder/pkg/gen/controller)
 godocs for reference documentation on controller annotations.
 
 
 {% sample lang="go" %}
 ```go
-// +kubebuilder:controller:group=workloads,version=v1alpha1,kind=ContainerSet,resource=containersets
-// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:informers:group=apps,version=v1,kind=Deployment
 type ContainerSetController struct {
-	args.InjectArgs
-    containersetrecorder record.EventRecorder
+	client.Client
+	scheme *runtime.Scheme
 }
 
-func ProvideController(arguments args.InjectArgs) (
-	    *controller.GenericController, error) {
-    bc := &ContainerSetController{
-		InjectArgs: arguments,
-        containersetrecorder: arguments.CreateRecorder(
-        	"ContainerSetController"),
-    }
+func Add(mgr manager.Manager) error (
+	// Create a new Controller
+	c, err := controller.New("containerset-controller", mgr,
+		controller.Options{Reconciler: &ContainerSetController{
+			Client: mgr.GetClient(),
+			scheme: mgr.GetScheme(),
+	}})
+	if err != nil {
+		return err
+	}
 
-    gc := &controller.GenericController{
-        Name: "ContainerSetController",
-        Reconcile: bc.Reconcile,
-        InformerRegistry: arguments.ControllerManager,
-    }
+	// Watch for changes to ContainerSet
+	err = c.Watch(&source.Kind{Type: &crewv1.FirstMate{}}, &handler.EnqueueRequestForObject{})
```

Comment:
Is `Type: &crewv1.FirstMate{}` for `ContainerSet`?  I think you want to use `&v1alpha1.ContainerSet`

Reply:
Yes